webapp/src/dogma/features/project/RestoreProject.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  Button,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { useRestoreProjectMutation } from 'dogma/features/api/apiSlice';
+import { createMessage } from 'dogma/features/message/messageSlice';
+import ErrorHandler from 'dogma/features/services/ErrorHandler';
+import { useAppDispatch } from 'dogma/store';
+import { MdOutlineRestore } from 'react-icons/md';
+
+export const RestoreProject = ({ projectName }: { projectName: string }) => {
+  const { isOpen, onToggle, onClose } = useDisclosure();
+  const dispatch = useAppDispatch();
+  const [restoreProject, { isLoading }] = useRestoreProjectMutation();
+  const handleRestore = async () => {
+    try {
+      await restoreProject({ projectName }).unwrap();
+      dispatch(
+        createMessage({
+          title: 'Project restored.',
+          text: `Successfully restored ${projectName}`,
+          type: 'success',
+        }),
+      );
+      onClose();
+    } catch (error) {
+      dispatch(
+        createMessage({
+          title: `Failed to restore ${projectName}`,
+          text: ErrorHandler.handle(error),
+          type: 'error',
+        }),
+      );
+    }
+  };
+  return (
+    <>
+      <Button colorScheme="blue" leftIcon={<MdOutlineRestore />} size="sm" variant="ghost" onClick={onToggle}>
+        Restore
+      </Button>
+      <Modal isOpen={isOpen} onClose={onClose}>
```

Comment:
Suggestion: We may introduce a component for our `Modal` design pattern in future work.